data-prepper-plugins/opensearch/src/main/java/org/opensearch/dataprepper/plugins/sink/opensearch/OpenSearchSink.java

```diff
@@ -589,7 +591,18 @@ private void logFailureForDlqObjects(final List<DlqObject> dlqObjects, final Thr
     if (getFailurePipeline() != null) {
       List<Record<Event>> records = new ArrayList<>();
       for (DlqObject dlqObject : dlqObjects) {
-        if (dlqObject.getEvent() != null) {
+        Event event = dlqObject.getEvent();
+
+        if (event != null) {
+          Map<String, Object> failureMetadata = new HashMap<>();
+          failureMetadata.put("pluginId", dlqObject.getPluginId());
+          failureMetadata.put("pluginaName", dlqObject.getPluginName());
+          failureMetadata.put("pipelineName", dlqObject.getPipelineName());
+          failureMetadata.put("message",  ((FailedDlqData) dlqObject.getFailedData()).getMessage());
+          failureMetadata.put("status", ((FailedDlqData) dlqObject.getFailedData()).getStatus());
+          failureMetadata.put("index", ((FailedDlqData) dlqObject.getFailedData()).getIndex());
+          failureMetadata.put("indexId", ((FailedDlqData) dlqObject.getFailedData()).getIndexId());
+          event.put("_failure_metadata", failureMetadata);
```

Comment:
I'd like to make this easier for plugin authors. This will quickly create tech debt if we continue this pattern.

Maybe a method on `Event`? 

```
event.setFailureMetadata(failureMetadata);
```

Or a factory somehow?